gravitee-management-api-rest/src/main/java/io/gravitee/management/rest/resource/ApiResource.java

```diff
@@ -118,14 +127,45 @@ private void setPermission(ApiEntity api) {
     @DELETE
     @Role({RoleType.OWNER, RoleType.TEAM_OWNER})
     public Response delete() {
-        apiService.findById(api);
 
         permissionService.hasPermission(getAuthenticatedUser(), api, PermissionType.EDIT_API);
 
         apiService.delete(api);
+        
         return Response.noContent().build();
     }
-
+    
+    @POST
+    @Role({RoleType.OWNER, RoleType.TEAM_OWNER})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("deploy")
+    public Response deployAPI() {
+    	permissionService.hasPermission(getAuthenticatedUser(), this.api, PermissionType.EDIT_API);
+    	
+    	try {
+    		apiService.deploy(api, getAuthenticatedUsername());
+		} catch (Exception e) {
+			return Response.status(Status.INTERNAL_SERVER_ERROR).entity("JsonProcessingException " + e).build();
+		}
+    	
+    	return Response.noContent().build();
+    }
+    
+    @GET
+    @Role({RoleType.OWNER, RoleType.TEAM_OWNER})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("isSynchronized")
```

Comment:
I'm not a big fan of this kind of URI for a rest API. Why not just @Path("state") ?